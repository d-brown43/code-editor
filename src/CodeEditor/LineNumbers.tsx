import * as React from "react";
import styles from './LineNumbers.module.scss';
import {generateRange} from "../helpers";

type LineNumbers = {
    program: string;
};

const LineNumbers = ({program}: LineNumbers) => {
    const lineCount = program.split('\n').length;

    return (
        <div className={styles.lineNumbers}>
            {generateRange(lineCount, 1).map((index) => {
                return (
                    <span key={index}>{index}</span>
                )
            })}
        </div>
    );
};

export default LineNumbers;
