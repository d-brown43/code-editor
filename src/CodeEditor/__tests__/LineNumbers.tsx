import LineNumbers from "../LineNumbers";
import {shallow} from "enzyme";
import * as React from "react";

it('matches snapshot', () => {
    const lineNumbers = shallow(
        <LineNumbers
            program={`
            HelloWorld
            `}
        />
    );
    expect(lineNumbers).toMatchSnapshot();
});
