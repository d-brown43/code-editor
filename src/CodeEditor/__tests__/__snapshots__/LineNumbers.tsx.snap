// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="lineNumbers"
>
  <span
    key="1"
  >
    1
  </span>
  <span
    key="2"
  >
    2
  </span>
  <span
    key="3"
  >
    3
  </span>
</div>
`;
